### DISCUSSION TITLE
Welcome to APBS Discussions!

### DISCUSSION THREAD

**Discussion by sobolevnrm** [Electrostatics/apbs#133] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aaomoniyi** [comment]
I am trying to compile apbs on ubuntu 20.04.2 and it fails for me:
I followed the tutorial https://apbs.readthedocs.io/en/latest/getting/source.html

git clone https://github.com/Electrostatics/apbs
cd apbs
git submodule init
git submodule update
Building the code - advanced

[ 14%] Building CXX object src/CMakeFiles/TABIPBlib.dir/molecule.cpp.o
cd /home/akinyemi/apbs/APBS_BUILD_DIR/bem-prefix/src/bem-build/src && /usr/bin/c++ -DTABIPB_APBS -I/home/akinyemi/apbs/externals/bem/src/../../../src -I/home/akinyemi/apbs/externals/bem/src/../../../build/src -I/home/akinyemi/apbs/externals/bem/src/../../../externals/fetk/maloc/src/base -I/home/akinyemi/apbs/externals/bem/src/../../../externals/fetk/maloc/src/vsys -I/home/akinyemi/apbs/externals/bem/src/../../../externals/fetk/maloc/src/vsh -I/home/akinyemi/apbs/externals/bem/src/../../../externals/fetk/maloc/src/psh -O2 -g -DNDEBUG -O3 -MD -MT src/CMakeFiles/TABIPBlib.dir/molecule.cpp.o -MF CMakeFiles/TABIPBlib.dir/molecule.cpp.o.d -o CMakeFiles/TABIPBlib.dir/molecule.cpp.o -c /home/akinyemi/apbs/externals/bem/src/molecule.cpp
In file included from /home/akinyemi/apbs/externals/bem/src/molecule.h:13,
                 from /home/akinyemi/apbs/externals/bem/src/molecule.cpp:12:
/home/akinyemi/apbs/externals/bem/src/../../../src/generic/valist.h:65:10: fatal error: apbscfg.h: No such file or directory
   65 | #include "apbscfg.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[6]: *** [src/CMakeFiles/TABIPBlib.dir/build.make:90: src/CMakeFiles/TABIPBlib.dir/molecule.cpp.o] Error 1
make[6]: Leaving directory '/home/akinyemi/apbs/APBS_BUILD_DIR/bem-prefix/src/bem-build'
make[5]: *** [CMakeFiles/Makefile2:100: src/CMakeFiles/TABIPBlib.dir/all] Error 2
make[5]: Leaving directory '/home/akinyemi/apbs/APBS_BUILD_DIR/bem-prefix/src/bem-build'
make[4]: *** [CMakeFiles/Makefile2:107: src/CMakeFiles/TABIPBlib.dir/rule] Error 2
make[4]: Leaving directory '/home/akinyemi/apbs/APBS_BUILD_DIR/bem-prefix/src/bem-build'
make[3]: *** [Makefile:169: TABIPBlib] Error 2
make[3]: Leaving directory '/home/akinyemi/apbs/APBS_BUILD_DIR/bem-prefix/src/bem-build'
make[2]: *** [CMakeFiles/bem.dir/build.make:86: bem-prefix/src/bem-stamp/bem-build] Error 2
make[2]: Leaving directory '/home/akinyemi/apbs/APBS_BUILD_DIR'
make[1]: *** [CMakeFiles/Makefile2:421: CMakeFiles/bem.dir/all] Error 2
make[1]: Leaving directory '/home/akinyemi/apbs/APBS_BUILD_DIR'
make: *** [Makefile:166: all] Error 2

the cmake build was successful but the VERBOSE=1 make -j 1 was giving the error above. 
plsease help!

**Reply by sobolevnrm** [reply]
@intendo would you be able to help with this?  Thanks!

**Comment by intendo** [comment]
Adding in @nsoblath since he has been working with the build system.

@aaomoniyi , can you run the command to see if the `apbscfg.h` is being built:
`find /home/akinyemi/apbs/externals -name apbscfg.h`

If the command returns, I think it should be in `/home/akinyemi/apbs/build/src/apbscfg.h`. If it is you might try the following:
```
cd /home/akinyemi/apbs/build
cmake -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS -I/home/akinyemi/apbs/build/src" .
```

**Comment by aaomoniyi** [comment]
Thank you.
I will try it out
